10/vincent

Vincentqchen opened the pull request:
Issue 10:

I have implemented the use of real data now. It is dynamically updated so all the client has to do is update the csv file and run the backend database. I also had to refactor a lot of old code because they all hardcoded in the data.

Vincentqchen committed:
Gave a preliminary look and change to the code to allow for real data to be used

Vincentqchen committed:
Allowed for dynamic pathway implementation using real data

Refactored a lot of old code to allow for use with real data. It is also dynamically updated so all users have to do is update the csv file to update the web server

nishi7409 commented:
Closes #10

amontes261 committed:
Minor cosmetic changes to the footer

footer content changes

Vincentqchen committed:
Finished up my issue

Fixed some bugs and recoded parts of the application to work with my updates

Vincentqchen committed:
update

Vincentqchen committed:
Added colored categories for each course

nishi7409 commented:
Editing pathways (after saving the three courses) is broken

(just documenting it)

![image](https://user-images.githubusercontent.com/20250366/114994987-febf7d00-9eba-11eb-9f89-1056e840f5eb.png)

nishi7409 committed:
Merge branch 'master' into 10/Vincent

nishi7409 committed:
directory routing fixed

Vincentqchen committed:
Update

amontes261 committed:
Resolved Issue with Edit Pathway Action action

- Resolved main Edit Pathway issue holding up PR

- Attempted to make it so that the first course is selected upon selecting Edit Pathway– could NOT implement

- Made it so changing the first course while in edit mode clears the second and third courses; changing the second course in the same mode clears only the third course
- Fixed issue where the first expansion panel was cut off in edit pathway mode (was originally cut off by the progress bar)
- Fixed issue where pathway name could appear as NULL in My Pathways page

nishi7409 reviewed (changes requested):
Editing pathways still doesn't work

Clicking on the first course option on the progress bar displays all of the available options; however, clicking on the second or third course options do not do the same. They're dependent on the previous course's option.

nishi7409 commented:
Editing the first course option
![image](https://user-images.githubusercontent.com/20250366/116038777-82c9ef80-a687-11eb-89f0-301a05a3c9b9.png)

Editing the second course option
![image](https://user-images.githubusercontent.com/20250366/116038823-92e1cf00-a687-11eb-8da0-487a95841b8e.png)

Editing the third course option
![image](https://user-images.githubusercontent.com/20250366/116038841-9a08dd00-a687-11eb-971f-f05b2593eb7f.png)

nishi7409 committed:
Merge branch 'master' into 10/Vincent

nishi7409 committed:
chipset updates

nishi7409 committed:
removed footer

forgot to mention it before, but yes, got rid of the footer because we don't need it

Vincentqchen committed:
Update FirstCourses.vue

Vincentqchen committed:
Merge branch '10/Vincent' of https://github.com/nishi7409/HASSPathways into 10/Vincent

Vincentqchen committed:
Update FirstCourses.vue

Vincentqchen committed:
Update MyPathways.vue

nishi7409 commented:
finally

nishi7409 commented:
took a full month to do this, 👍
